### PR TITLE
Fix for typo scanner.antiCSFR should be scanner.antiCSRF

### LINKED ISSUE #7598: Misspelling in `config.xml` `scanner.antiCSFR` should be `scanner.antiCSRF`
### Describe the bug

Minor typo.

### Steps to reproduce the behavior

`grep CSFR ~/.ZAP/config.xml`

### Expected behavior

Just a spelling correction but may have larger ramifications since it's in the config file.

### Software versions

2.12.0

### Screenshots

_No response_

### Errors from the zap.log file

_No response_

### Additional context

_No response_

### Would you like to help fix this issue?

- [ ] Yes

### PR DESCRIPTION
**Pull request by sumeetpatil**
Fix #7598 

Signed-off-by: sumeet patil <sumeetp1991@gmail.com>

### CODE REVIEW

**Comment by thc202** [comment]
The old config needs to be updated.

**Comment by kingthorin** [comment]
The tags, conditions, and a new `updateFrom` method would be needed in:  https://github.com/zaproxy/zaproxy/blob/main/zap/src/main/java/org/parosproxy/paros/Constant.java

**Comment by kingthorin** [comment]
Here’s an example that might help: https://github.com/zaproxy/zaproxy/pull/5719/files

**Comment by sumeetpatil** [comment]
Thanks, @kingthorin for the references. I will change it now.

**Comment by thc202** [comment]
It needs to be updated always, the user might be setting that configuration with the `-config` argument.

File: zap/src/main/java/org/parosproxy/paros/Constant.java [hunk 1]
```diff
@@ -1140,6 +1142,16 @@ private static void upgradeFrom2_11_1(XMLConfiguration config) {
...
 
+    static void fixTypoForScannerAntiCSRF(XMLConfiguration config) throws ConfigurationException {
+        String oldKeyName = "scanner.antiCSFR"; // This is the typo we want to fix
+        String newKeyName = "scanner.antiCSRF";
+        if(config.containsKey(oldKeyName)){
+            config.setProperty(newKeyName, config.getProperty(oldKeyName));
+            config.clearProperty(oldKeyName);
+            config.save();
+        }
+    }
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/Constant.java:1153]: This should be done in `ScannerParam`.

**Comment by kingthorin** [comment]
Thanks @sumeetpatil

File: zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java [hunk 2]
```diff
@@ -311,6 +312,14 @@ protected void parse() {
         }
     }
 
+    private void fixTypoForScannerAntiCSRF() {
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java:315]: Add to `removeOldOptions` and rename it to `migrateOldOptions`.

File: zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java [hunk 3]
```diff
@@ -311,6 +312,14 @@ protected void parse() {
...
+        final String oldKey = "scanner.antiCSFR"; // This is the typo we want to fix
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java:316]: Remove final and the comment.

File: zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java [hunk 4]
```diff
@@ -311,6 +312,14 @@ protected void parse() {
...
+        if(getConfig().containsKey(oldKey)){
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java]: This needs a `spotlessApply`.

File: zap/src/test/java/org/parosproxy/paros/core/scanner/ScannerParamUnitTest.java [hunk 5]
```diff
@@ -185,4 +185,15 @@ void shouldSetScanJsonNullValuesToConfiguration(boolean value) {
         assertThat(
                 configuration.getBoolean(ScannerParam.SCAN_NULL_JSON_VALUES), is(equalTo(value)));
     }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void fixTypoForScannerAntiCSRF(boolean scannerAntiCSRF) {
```
Comment (thc202) [inline zap/src/test/java/org/parosproxy/paros/core/scanner/ScannerParamUnitTest.java:191]: Use existing naming pattern.

File: zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java (hunk 2 above)
Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java:314]: Do this in `migrateOldOptions`, there's no need to have it in its own method (simple statements and will be called just from one place).

File: zap/src/test/java/org/parosproxy/paros/core/scanner/ScannerParamUnitTest.java [hunk 6]
```diff
@@ -185,4 +185,15 @@ void shouldSetScanJsonNullValuesToConfiguration(boolean value) {
...
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void shouldFixTypoForScannerAntiCSRF(boolean scannerAntiCSRF) {
+        // Given
+        configuration.setProperty("scanner.antiCSFR", scannerAntiCSRF);
+        // When
+        param.load(configuration);
+        // Then
+        assertThat(param.getHandleAntiCSRFTokens(), is(scannerAntiCSRF));
```
Comment (thc202) [inline zap/src/test/java/org/parosproxy/paros/core/scanner/ScannerParamUnitTest.java:197]: Should assert that the old key no longer exists and that the new one does.

File: zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java (hunk 2 above)
Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java:314]: Bump. The commits can also be squashed.

**Comment by sumeetpatil** [comment]
Hi @thc202 , I think it should be good now. Can you please revisit this PR now?

File: zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java [hunk 7]
```diff
@@ -311,10 +311,17 @@ protected void parse() {
         }
     }
 
-    private void removeOldOptions() {
-        final String oldKey = "scanner.deleteOnShutdown";
-        if (getConfig().containsKey(oldKey)) {
-            getConfig().clearProperty(oldKey);
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java:317]: Keep as is and just drop the final, use the same variable to fix the typo.

Comment (sumeetpatil) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/ScannerParam.java]: done

**Comment by sumeetpatil** [comment]
Sorry, missed running spotlessApply. @thc202 please rebuild.

**Review by thc202 — APPROVED** [review]

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Thanks

**Pull request merged**